src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -59,24 +59,54 @@ public class HubEndPoint<THub, TClient> : IInvocationBinder where THub : Hub<TCl
 
         public async Task OnConnectedAsync(ConnectionContext connection)
         {
-            await ProcessNegotiate(connection);
+            var output = Channel.CreateUnbounded<byte[]>();
+            var connectionContext = new HubConnectionContext(output, connection);
+
+            await ProcessNegotiate(connectionContext);
```

Comment:
in other words 
- if the client requested Protobuf, but webSocketOptions.WebSocketMessageType was set to text (which is the default) the server will send binary data over text violating the protocol. 
- if the user sets webSocketOptions.WebSocketMessageType to binary but the client requests JSon we will send text messages over binary which requires additional processing on the client to convert to string
- with SSE it does not work at all because it is a text protocol. We can't base64 encode everything though since we don't want to base64 encode text messages